spotify_tensorflow/freeze_graph.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import logging
+
+import tensorflow as tf
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class FreezeGraph(object):
+
+    @classmethod
+    def session(cls, session, path, network):
+        """
+        Freeze a graph by taking a session and a network and storing
+        the results into a pb file at the given path. This function wil convert
+        variables to constants which is necessary for JVM serving.
+
+        :param session: TF Session
+        :param path: Where the graph will be written
+        :param network: Tensor representing a network(Usually a prediction)
+        :return: Path to the written graph
+        """
+        input_graph_def = tf.get_default_graph().as_graph_def()
+
+        logger.info("Freezing model")
```

Comment:
I wonder whether we could add `Freezing model at <time>` and once it's written down, `Model frozen at <path>, process took: <duration>`. What do you think?